Ether.Core/Reporters/WeeklyStatusReporter.cs

```diff
@@ -0,0 +1,230 @@
+            if (workItem.Updates == null || !workItem.Updates.Any())
+                return 0.0f;
+
+            var activeTime = 0.0F;
+            var isActive = false;
+            var assignedToTeam = false;
+            DateTime? lastActivated = null;
+            foreach (var update in workItem.Updates)
+            {
+                var isActivation = !update.State.IsEmpty && update.State.NewValue == WorkItemStates.Active;
+                var isOnHold = !update.State.IsEmpty && update.State.NewValue == WorkItemStates.New;
+                var isResolved = !update.State.IsEmpty && (update.State.NewValue == WorkItemStates.Resolved || update.State.NewValue == WorkItemStates.Closed);
+                var isCodeReview = !update.Tags.IsEmpty && WorkItemTags.ContainsTag(update.Tags.NewValue, WorkItemTags.CodeReview) || update.Relations?.Added != null 
+                                   && update.Relations.Added.Any(i => !string.IsNullOrWhiteSpace(i.Name) && i.Name.Equals("Pull Request", StringComparison.InvariantCultureIgnoreCase));
+                var isBlocked = !update.Tags.IsEmpty &&
+                    (WorkItemTags.ContainsTag(update.Tags.NewValue, WorkItemTags.Blocked) || WorkItemTags.ContainsTag(update.Tags.NewValue, WorkItemTags.OnHold));
+                var isUnBlocked = !isBlocked && !update.Tags.IsEmpty &&
+                    (WorkItemTags.ContainsTag(update.Tags.OldValue, WorkItemTags.Blocked) && !WorkItemTags.ContainsTag(update.Tags.NewValue, WorkItemTags.Blocked) ||
+                     WorkItemTags.ContainsTag(update.Tags.OldValue, WorkItemTags.OnHold) && !WorkItemTags.ContainsTag(update.Tags.NewValue, WorkItemTags.OnHold));
+
+                if (!assignedToTeam && !string.IsNullOrWhiteSpace(update.AssignedTo.NewValue))
+                {
+                    assignedToTeam = Input.Members.Any(m => update.AssignedTo.NewValue.Contains(m.Email));
+                    if(isActive) lastActivated = update.ChangedDate;
+                }
+
+                if (isActive && (isOnHold || isBlocked))
+                {
+                    isActive = false;
+                    if (lastActivated != null && assignedToTeam)
+                        activeTime += CountBusinessDaysBetween(lastActivated.Value, update.ChangedDate);
+                }
+                else if ((isActivation && !isBlocked) || isUnBlocked)
+                {
+                    lastActivated = update.ChangedDate;
+                    isActive = true;
+                }
+
+                else if (isActive && (isResolved || isCodeReview))
+                {
+                    if (lastActivated != null && assignedToTeam)
+                        activeTime += CountBusinessDaysBetween(lastActivated.Value, update.ChangedDate);
+                    break;
+                }
+            }
+
+            return activeTime;
+        }
+
+        public static float CountBusinessDaysBetween(DateTime firstDay, DateTime lastDay, params DateTime[] holidays)
```

Comment:
If this is the same one as in the ETA reporter, can you move it to util class and reuse, please?